### PR TITLE
[GATING] - Create and Edit Group UI

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: 
https://github.com/hicommonwealth/commonwealth/issues/4991
https://github.com/hicommonwealth/commonwealth/issues/5150

## Description of Changes
- Added "create group" and "update group" page UI's and forms with dynamic validation and error states. Accessible at `/:scope/members/groups/create` and `/:scope/members/groups/:id/update`.
- Only admins of a community can access these pages
- Connected mock API (TODO: Connect the real API once developed)
- Added new components, Select List from https://github.com/hicommonwealth/commonwealth/issues/5085 and a fused-in Radio label with text input (its not generic)

## "How We Fixed It"
This is a new feature.

## Test Plan
Since this is a new feature

- Visit the pages (`/:scope/members/groups/create` and `/:scope/members/groups/:id/update`) and confirm the page UI matches the expected UI defined in https://github.com/hicommonwealth/commonwealth/issues/4991
and https://github.com/hicommonwealth/commonwealth/issues/5150.
- Visit the pages (`/:scope/members/groups/create` and `/:scope/members/groups/:id/update`) and confirm the product copy/ux matches the expected copy/ux defined in https://github.com/hicommonwealth/commonwealth/issues/4991
and https://github.com/hicommonwealth/commonwealth/issues/5150.
- Visit the pages (`/:scope/members/groups/create` and `/:scope/members/groups/:id/update`) and confirm the engineering behavior matches the expected behavior defined in https://github.com/hicommonwealth/commonwealth/issues/4991
and https://github.com/hicommonwealth/commonwealth/issues/5150.

## Deployment Plan
N/A

## Other Considerations
TODO: Connect the real API once developed

### CODE REVIEW

**Comment by mzparacha** [comment]
### Update

Some UI components don't exist as part of the design system, so I will have to make them. Will add them in this PR.

image

image


cc: @jnaviask @CowMuon

**Comment by jnaviask** [comment]
@mzparacha do you have separate tickets for the components required? also cc @jessmart1213

**Comment by mzparacha** [comment]
I have also noticed that we don't use form control (input type radio) for radio buttons, I think this can be a separate ticket (lmk if I should create one or manage it here)

The suggestion is to use HTML form controls (instead of just UI elements), for our form control UI. This is more consistent with usage and will also help with "better form" validation stuff.

cc: @jnaviask @Miaplacidus @masvelio

**Comment by mzparacha** [comment]
Ready for design/product review.

cc: @zakhap @jnaviask @jessmart1213.

File: packages/commonwealth/client/scripts/state/api/groups/createGroup.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/groups/createGroup.ts:1]: TODO: complete this once api is done

File: packages/commonwealth/client/scripts/state/api/groups/editGroup.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/groups/editGroup.ts:1]: TODO: complete this once api is done

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_radio_button.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_radio_button.tsx:1]: I updated the radio button implementation to use form control @Miaplacidus

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:1]: The `GroupForm` is reuseable, basically a plug and play at this moment. Its used for both the create/update group pages

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:34]: Ideally we could move `CWRequirementsRadioButton` into a separate file, this is the radio button with the text input in label.

I wouldn't recommend to move it to separate file, as its more meaningful here.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:147]: .

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:167]: .

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:312]: Todo: add breadcrumb once its finished

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx:10]: I added these constants in this PR, but seems like they are also used in some other places but hardcoded. 

I think it a good idea to move this to a global constant, but maybe that's something for later.

cc: @masvelio

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/validations.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/validations.ts:7]: I would move these to a more global area once we get more forms that use these same messages, but to reduce noise i won't over abstract for now.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/index.tsx:54]: This prop will fill the form fields with the values in the `initialValues` object. Very helpful when the form is used to update existing data (which we can pre-fill in the form very easily)

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.scss
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.scss:43]: change the name of the file

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx:34]: you created file with constants, can you move them there?

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx:161]: change file name to RequirementSubForm.tsx

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx:81]: use clsx for dynamic classes

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts:1]: nit: Not sure if we have a rule for that but most of enums I see in the codebase do not use uppercase notation, rather they are like this:

```
enum BulkBalancesErrors {
  InvalidToken = 'Invalid token',
}
```

**Review by masvelio — COMMENTED** [review]
left some comments, good job in general! Thanks for applying new way of handling forms

**Comment by masvelio** [comment]
image

some mobile width issues with buttons

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx:7]: generally I think this might be a new CWBanner (alert) component but maybe design team had different desired. Imo we should be reusing components from the design system

File: packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx:166]: ah nice spot, yes. Will add that.

File: packages/commonwealth/client/scripts/state/api/groups/createGroup.ts
Comment (mzparacha) [reply packages/commonwealth/client/scripts/state/api/groups/createGroup.ts:26]: It depends on the api changes. I believe once the api changes are done then these can be followed up

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_radio_button.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_radio_button.tsx:64]: I tried to follow the same visuals, dimensions, but yes we should review those too.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.tsx:33]: Done

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/index.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/index.tsx:70]: Done

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.scss
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.scss:1]: We actually don't need an empty file, I was just following the convention we do with new files

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx]: TBH, i don't think this will be "contexful" if we move there, these are not actually constants, but the options objects for the select list

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:79]: Done

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:78]: Done

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:121]: We only want to run this on the mount, so I disabled the rule. Adding in the dependency array will recompute everytime dependency changes

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts:1]: Hm, should i update the one's i added to be like this?

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:307]: lets followup [here](https://github.com/hicommonwealth/commonwealth/pull/5045#discussion_r1344133284)

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:37]: 1. replied [here](https://github.com/hicommonwealth/commonwealth/pull/5045#discussion_r1344108739)
2. yes its only used in this place and is basically baked in (non-generic).

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx]: > Imo we should be reusing components from the design system

Hm, I couldn't find a similar one. Do you know if we already have a similar one?

**Comment by mzparacha** [comment]
re: https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1745063961

I don't think we have mobile designs yet. but yeah I can fix the overflow. Lmk if I should do that.

cc: @zakhap @jessmart1213

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:152]: > I am ok with this "hack". Don't even know if this is a hack ;)

In terms of our "better form" design, I would consider it a hack, the purpose of the newer forms design was to make life easy but these kind of ad-hoc validations will do the opposite. To reduce friction, I believe it can go in for now.

> The only thing I would improve here is that quite a lot of the code is being repeated 3 times. Maybe you could create some util function in separate file or custom hook that will do the heavy lifting, so this component won't be so polluted with custom code?

done

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx]: Added a const

**Comment by mzparacha** [comment]
I will fix merge conflicts after we finish the eng review

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss:1]: replied [here](https://github.com/hicommonwealth/commonwealth/pull/5045#discussion_r1346193719)

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:42]: I am not gonna argue here because I am also not sure when we should and when we should not use CW prefix before the component name. Maybe @gdjohnson would know or document that if possible?

Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:42]: this file has 500 lines of code and in this situation we really should move utils and other components to separate files. It does not cost us anything to create a new file with new component.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:141]: I know, but what is the case when the `initialValues` are being changed? Initial value should be unchanged, the same for cwRequiremnetsLabelInputField. Disabling this rule is last resort and in this case passing second argument should not break the functionality.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/constants.ts:1]: up to you

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss:1]: we do not need stubs/empty files. We should create them when the styles should be added

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.scss
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Create/index.scss]: yeah, lets get rid of empty files

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/RequirementSubForm/index.tsx]: makes sense

**Comment by masvelio** [comment]
> re: [#5045 (comment)](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1745063961)
> 
> I don't think we have mobile designs yet. but yeah I can fix the overflow. Lmk if I should do that.
> 
> cc: @zakhap @jessmart1213

We do not have designs for most of the features, but even though we do not have it, it does not mean we should not care about the mobile. We should just make sure that content fits the screen, either being full with or wrap it to the next line. This is low hanging fruit.

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/TopicGatingHelpMessage/index.tsx]: As I said, [CWBanner (alert)](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=793%3A11228&mode=dev). It is not identical so this would need require feedback from design, if we should reuse alert component or create custom one as you did

**Review by masvelio — APPROVED** [review]
left some small comments but most of the work looks great!

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/Update/UpdateCommunityGroupPage.scss:1]: I see, didn't knew this was the convention. I will update it shortly.

**Comment by mzparacha** [comment]
> We do not have designs for most of the features, but even though we do not have it, it does not mean we should not care about the mobile. We should just make sure that content fits the screen, either being full with or wrap it to the next line. This is low hanging fruit.

I agree, also I remember some fluid vs breakpoint responsiveness touchdown in some meeting last week. Maybe in cases like this we will go fluid responsive?

@masvelio

**Comment by rbennettcw** [comment]
For `evm_chain_id`, you need to first get `Chain.chain_node_id`, then get the `ChainNode` for that ID. The value you want is `ChainNode.eth_chain_id`.

I believe all the data is fetched in the status route.

**Comment by mzparacha** [comment]
> For `evm_chain_id`, you need to first get `Chain.chain_node_id`, then get the `ChainNode` for that ID. The value you want is `ChainNode.eth_chain_id`.
> 
> I believe all the data is fetched in the status route.

This is for point 3, right?

**Comment by mzparacha** [comment]
> For `evm_chain_id`, you need to first get `Chain.chain_node_id`, then get the `ChainNode` for that ID. The value you want is `ChainNode.eth_chain_id`.
> 
> I believe all the data is fetched in the status route.

This is for point 3, right?

**Comment by rbennettcw** [comment]
`required_requirements` is not implemented on the backend. It was not clearly defined what exactly that property should do.

For topics, can you show a screenshot of what the UI looks like so I can understand what data you need?

All chains have a `Chain.default_symbol` property that you should be able to use.

Regarding validation, if you get errors, my recommendation is to debug them one at a time. The JSON schema supports multiple possible formats and the parser cannot guess which format you want, so it won't always provide meaningful errors.

Check out `requirementsTypes.ts` to see the different data structures you can use for the source.

**Comment by mzparacha** [comment]
> For topics, can you show a screenshot of what the UI looks like so I can understand what data you need?


image

This is a card, that would fill the group data. and I am using this sample schema to populate it (again this schema is just a sample)

```
const groupData = {
  isJoined: true,
  groupName: 'Hedgies Pod',
  groupDescription:
    'Lorem ipsum dolor sit amet consectetur. Urna et at velit platea sagittis feugiat gravida augue. Et mi quam mattis nisl proin scelerisque ultricies enim. Purus in eget sed rutrum vulputate in fermentum. Quam etiam nunc tristique nibh arcu tempor.',
  requirementsToFulfill: 'ALL',
  requirements: [
    {
      requirementAmount: '100',
      requirementChain: 'Ethereum',
      requirementContractAddress: '0x9AEFd6d4feF5b70119FC194D81EbAD761A3269C4',
      requirementType: 'ERC-20',
      requirementCondition: 'Equal to',
    },
    {
      requirementAmount: '50',
      requirementChain: 'Ethereum',
      requirementType: 'Evm Base Tokens',
      requirementCondition: 'Equal to',
    },
  ],
  topics: ['Call updates', 'Hedgies'],
};
```

**Comment by mzparacha** [comment]
@rbennettcw bump on [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769204902) message.

Can you also confirm point 1 from [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769116790) comment

**Comment by mzparacha** [comment]
Also, me and Ryan has some discussion on this in dm's, just adding for visibility (this is again for the gating group form "requirements" section)

A requirement sub form looks like this

image


and the condition field can have these values
image


but the backend only supports the "more than" option, so atm we don't send this value as a key to backend (per ryan)

What should be the proper flow here, to be specific
1. Since the backend automatically sets "condition = more than", should we remove this field from the frontend?
2. If we don't follow option 1, do we disable this field and set "more than" as the default selected value?
3. Do we add a new request body param to the post/put /groups endpoint to store this value from frontend?

cc: @rbennettcw @jnaviask @zakhap

**Comment by mzparacha** [comment]
### Status

Summarizing the discussion from above after today's updates

#### Pending Eng/Product things

1. The `required_requirements` field is not returned by the API - we will need this in the API response schema for this field in the create/update group page. (previous context [1](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769116790), [2](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769192680), [3](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1770991215)) (@zakhap / @rbennettcw)

image


per @rbennettcw 
> required_requirements is not implemented on the backend. It was not clearly defined what exactly that property should do.


What should be the next steps for this? Should we just remove this field from frontend? Or maybe update API to return this key? @jnaviask @zakhap 

2. The condition field has these options but we only use the "more than" option on the API side. What should we do on the frontend? ([previous context](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1771008526))
image


3. The ethChainId of some chains is `null` please see these chains list (this is the exact list we use in frontend)

image


```
[
    {
        "value": 2020,
        "label": "Axie-Infinity"
    },
    {
        "value": null,
        "label": "Cosmos"
    },
    {
        "value": 1,
        "label": "Ethereum"
    },
    {
        "value": null,
        "label": "Injective"
    },
    {
        "value": null,
        "label": "Near"
    },
    {
        "value": null,
        "label": "Polkadot"
    },
    {
        "value": 1,
        "label": "Polygon"
    },
    {
        "value": null,
        "label": "Solana"
    }
]
```

What should we do in this case? I got these from app.config.chains[] => chain.ChainNode.ethChainId


4. Per design the "topics" were editable (after group creation) but It seems the API doesn't support it.
image


Can you please confirm if they are editable or not? @zakhap @jnaviask 




cc: @jnaviask @CowMuon @rbennettcw @zakhap

**Comment by jnaviask** [comment]
@mzparacha re 3, some of the chains will have cosmos_chain_id instead of eth_chain_id. As for the other Qs, I defer to Zak/Jessica.

**Comment by mzparacha** [comment]
I had a look but these chains don't have a cosmos or an eth chain id => Cosmos, Injective, Near, Polkadot, Solana

Can it be because of local database?

@jnaviask

**Comment by zakhap** [comment]
> 1. Since the backend automatically sets "condition = more than", should we remove this field from the frontend?

cc: @jnaviask + @rbennettcw: we will want to expand the options here to include `> < ≤ ≥ =` in the future. There are clear on-chain usecases for all of them. But based on current backend implementation, that seems like an unreasonable ask for V1. 

@rbennettcw: is the current backend implementation "more than" or "more than or equal to" ??

> 2. If we don't follow option 1, do we disable this field and set "more than" as the default selected value?

For the moment, we should default to "more than" in the dropdown and make the dropdown disabled. 

> The ethChainId of some chains is null please see these chains list (this is the exact list we use in frontend). What should we do in this case? I got these from app.config.chains[] => chain.ChainNode.ethChainId

Ok. As @jnaviask mentioned above, ChainNodes store their chainIds in two different locations depending on the Blockchain Ecosystem. For ERC20 + ERC721 + EVM Base Token requirements, the options available in the Chain selector dropdown should only be ChainNodes that have an ethChainId (not null). For Cosmos Base Token requirement, the options available should only be ChainNodes with "cosmosChainId". 

> Per design the "topics" were editable (after group creation) but It seems the API doesn't support it. Can you please confirm if they are editable or not?

cc @rbennettcw + @jnaviask: it is **absolutely imperative** that admins can change the Topics that a Group can post in.

**Comment by mzparacha** [comment]
After merging master (having existing gating code), some issues came which I have fixed above. Now for these small things below I will create a followup ticket and merge this PR

- Required Requirements value
- Eth/Cosmos chain id's and the dropdown
- Make Topics Editable

**Review by kurtisassad — APPROVED** [review]
LGTM. Since its a huge PR I will manually merge it past the code quality recommendations.

**Pull request merged**
